src/main/java/org/osc/manager/rest/server/api/DomainApis.java

```diff
@@ -40,19 +49,57 @@
 public class DomainApis {
 
     private static final Logger logger = Logger.getLogger(DomainApis.class);
+    private EntityManager em;
+    private TransactionControl txControl;
+
+    /**
+     * init the DB
+     *
+     *
+     */
+    public void init(EntityManager em, TransactionControl txControl) throws Exception {
+        this.em = em;
+        this.txControl = txControl;
+    }
 
     /**
      * Creates the Policy for a given domain
      *
      * @return policy
      */
-    @Path("/{domainid}")
+    @Path("/{domainid}/policies")
     @POST
-    public PolicyEntity createPolicy(@PathParam("domainid") Long domainid, PolicyEntity entity) {
+    public String createPolicy(@PathParam("domainid") Long domainid, PolicyEntity entity) {
+
+        logger.info("Creating Policy Entity...:" + entity.getName());
+
+        return this.txControl.required(new Callable<PolicyEntity>() {
+
+            @Override
+            public PolicyEntity call() throws Exception {
 
-        logger.info("Creating Policy Entity...");
-        //TODO Sudhir: Add db calls here
-        return null;
+                DomainEntity result = em.find(DomainEntity.class, domainid);
+                if (result == null) {
+                    throw new Exception("Domain Entity does not exists...");
+                    //TODO - to add RETURN 404 error:Sudhir
+                }
+                CriteriaBuilder criteriaBuilder = em.getCriteriaBuilder();
+                CriteriaQuery<PolicyEntity> query = criteriaBuilder.createQuery(PolicyEntity.class);
+                Root<PolicyEntity> r = query.from(PolicyEntity.class);
+                query.select(r).where(criteriaBuilder.and(criteriaBuilder.equal(r.get("name"), entity.getName())));
+
+                List<PolicyEntity> policyresult = em.createQuery(query).getResultList();
+                if (!policyresult.isEmpty()) {
+                    throw new Exception("Policy Entity name already exists...:");
+                    //TODO - to add RETURN 404 error:Sudhir
+                }
+                PolicyEntity policy = new PolicyEntity();
+                policy.setName(entity.getName());
+                policy.setDomain(result);
+                em.persist(policy);
```

Comment:
can it simply be em.presist(entity) without the need to create a new policy object?

Reply:
ok